### PR TITLE
Add Clarity

### PR DESCRIPTION
**Pull request by Jinnie**
Project Clarity is an open source design system that brings together UX guidelines, an HTML/CSS framework, and Angular components. It's been started and supported by VMware, and is used in the majority of their products.
https://clarity.design/
https://github.com/vmware/clarity (4500+ Stars)
https://www.npmjs.com/package/@clr/angular (tens of thousands of weekly downloads)

### CODE REVIEW

File: src/apps.json [hunk 1]
```diff
@@ -1785,6 +1785,24 @@
       },
       "website": "http://ckan.org/"
     },
+    "Clarity": {
+      "cats": [
+        18
+      ],
+      "html": [
+        "<clr-main-container",
+        "<link [^>]*href=\"[^\"]*clr-angular(?:\\.umd)?(?:\\.min)?\\.js",
```
Comment (jvoisin) [inline src/apps.json]: This belongs in a `script` detection.

File: src/apps.json [hunk 2]
```diff
@@ -1785,6 +1785,24 @@
...
+        "<link [^>]*href=\"[^\"]*clr-ui(?:\\.min)?\\.css"
```
Comment (jvoisin) [inline src/apps.json]: What about something like `"<link[^>]+href=[^\"]+clr-ui(?:\\.min)?\\.css"` instead?

Comment (Jinnie) [reply src/apps.json]: Hi @jvoisin
Thanks for the quick reviews :) I have a few questions about this comment though:
How will your pattern match `<link href="clr-ui.min.css"`?
We directly exclude " after the href= and then we force any non-" char before clr-ui
Also, the `<a [^>]*href=\"` pattern is clearly advertised on the https://www.wappalyzer.com/docs/specification page and I also find it no less meaningful than `<a[^>]+href`
What makes you prefer the latter?

Comment (jvoisin) [reply src/apps.json]: Oups, you're right, `href=[^\"]` doesn't work, my bad.

I do prefer the `+` operator to `*` for performances concerns.

**Comment by Jinnie** [comment]
Travis fails after complaining about an icon that has nothing to do with my commit :(
`Error: Stray file (no associated app): src/icons/UserEngage.png`

Hmm... it seems that I depend on this one #2628

File: src/apps.json (hunk 1 above)
Comment (Jinnie) [reply src/apps.json]: Thanks! Fixed

File: src/apps.json (hunk 2 above)
Comment (Jinnie) [reply src/apps.json]: I can change it if you insist, but I don't think one is more greedy than the other. It all depends on what precedes them.
If you can point any reference article that discusses performance of + vs *, I'd be grateful. I'm not trolling :) just Googling for + and * is a bit... hard and I'm really interested into the topic.

Comment (Jinnie) [reply src/apps.json]: Any update on this?
Change it or leave it? I wouldn't like to miss the next mass merge.
Thanks!

Comment (Jinnie) [reply src/apps.json]: Resolving due to inactivity. Request change again if you think it's important.

**Comment by AliasIO** [comment]
Thanks!

**Pull request merged**